src/main/java/seedu/us/among/logic/endpoint/PostRequest.java

```diff
@@ -0,0 +1,41 @@
+package seedu.us.among.logic.endpoint;
+
+import java.io.IOException;
+
+import org.apache.http.client.methods.HttpPost;
+
+import seedu.us.among.model.endpoint.Endpoint;
+import seedu.us.among.model.endpoint.Response;
+
+/**
+ * Contains the logic for sending post requests.
+ */
+public class PostRequest extends Request {
+
+    private final Endpoint endpoint;
+
+    /**
+     * Constructor for PostRequest.
+     *
+     * @param endpoint endpoint to make API call on
+     */
+    public PostRequest(Endpoint endpoint) {
+        super(endpoint.getMethod().getMethodType(), endpoint.getAddress().value);
+        this.endpoint = endpoint;
+    }
+
+    /**
+     * Executes the API call with a get request.
```

Comment:
Small typo in the comment, should be a post request

Reply:
Thank you for catching that, it's fixed! :)